aspnetcore/mvc/razor-pages/index.md

```diff
@@ -69,398 +67,254 @@ The associations of URL paths to pages are determined by the page's location in
 
-        [Required]
-        public string Email { get; set; }
-    }
-}
-```
+The *Pages/Create.cshtml* view file:
 
-The *MyApp/Pages/Contact.cshtml* file:
+[!code-html[main](index/sample/RazorPagesContacts/Pages/Create.cshtml)]
 
-```html
-@page
-@using MyApp
-@using Microsoft.AspNetCore.Mvc.RazorPages
-@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
-@inject ApplicationDbContext Db
-
-@functions {
-    [BindProperty]
-    public Contact Contact { get; set; }
-    
-    public async Task<IActionResult> OnPostAsync()
-    {
-        if (ModelState.IsValid)
-        {
-            Db.Contacts.Add(Contact);
-            await Db.SaveChangesAsync();
-            return RedirectToPage();
-        }
-        
-        return Page();
-    }
-}
+The *Pages/Create.cshtml.cs* code-behind file for the view:
 
-<html>
-<body>
-    <p>Enter your contact info here and we will email you about 
-         our fine products!</p> 
-    <div asp-validation-summary="All"></div>
-    <form method="POST">
-      <div>Name: <input asp-for="Contact.Name" /></div>
-      <div>Email: <input asp-for="Contact.Email" /></div>
-      <input type="submit" />
-    </form>
-</body>
-</html>
-```
+[!code-cs[main](index/sample/RazorPagesContacts/Pages/Create.cshtml.cs?name=ALL)]
+
+By convention, the `PageModel` class is called `<PageName>Model` and is in the same namespace as the page. Not much change is needed to convert from a page using `@functions` to define handlers and a page using a `PageModel` class. 
+
+Using a `PageModel` code-behind file supports unit testing, but requires you to write an explicit constructor and class. Pages without `PageModel` code-behind files support runtime compilation, which can be an advantage in development.  <!-- review: advantage because you can make changes and refresh the browser without explicitly compiling the app -->
 
 The page has an `OnPostAsync` *handler method* which runs on `POST` requests (when a user posts the form). You can add handler methods for any HTTP verb. The most common handlers are:
 
-* `OnGet` to initialize state needed for the page.
+* `OnGet` to initialize state needed for the page. [OnGet](#OnGet) sample.
 * `OnPost` to handle form submissions. 
 
-The `Async` naming suffix is optional but is often used by convention. The code that's in `OnPostAsync` in the preceding example looks similar to what you would normally write in a controller. This is typical for pages. Most of the MVC primitives like model binding, validation, and action results are shared.
+The `Async` naming suffix is optional but is often used by convention. The code that's in `OnPostAsync` in the preceding example looks similar to what you would normally write in a controller. This is typical for Razor Pages. Most of the MVC primitives like [model binding](xref:mvc/models/model-binding), [validation](xref:mvc/models/validation), and action results are shared.  <!-- Review: Ryan, can we get a list of what is shared and what isn't? -->
```

Comment:
~~The code that's in `OnPostAsync` in the preceding example looks similar to what you would normally write in a controller.~~
The `OnPostAsync` code in the preceding example looks similar to what you would normally write in a controller.